src/Microsoft.AspNet.Mvc.TagHelpers/ImageTagHelper.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Hosting;
+using Microsoft.AspNet.Mvc.TagHelpers.Internal;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.Framework.Caching.Memory;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers
+{
+    /// <summary>
+    /// <see cref="ITagHelper"/> implementation targeting &lt;img&gt; elements that supports file versioning.
+    /// </summary>
+    /// <remarks>
+    /// The tag helper won't process for cases with just the 'src' attribute.
+    /// </remarks>
+    [TargetElement("img", Attributes = FileVersionAttributeName)]
+    public class ImageTagHelper : TagHelper
+    {
+        private static readonly string Namespace = typeof(ImageTagHelper).Namespace;
+
+        private const string FileVersionAttributeName = "asp-file-version";
+        private const string SrcAttributeName = "src";
+
+        private FileVersionProvider _fileVersionProvider;
+        
+        /// <summary>
+        /// Source of the image.
+        /// </summary>
+        /// <remarks>
+        /// Passed through to the generated HTML in all cases.
+        /// </remarks>
+        [HtmlAttributeName(SrcAttributeName)]
+        public string Src { get; set; }
+
+        /// <summary>
+        /// Value indicating if file version should be appended to the src urls.
+        /// </summary>
+        /// <remarks>
+        /// If <c>true</c> then a query string "v" with the encoded content of the file is added.
+        /// </remarks>
+        [HtmlAttributeName(FileVersionAttributeName)]
+        public bool? FileVersion { get; set; }
+
+        [Activate]
+        [HtmlAttributeNotBound]
+        public IHostingEnvironment HostingEnvironment { get; set; }
+
+        [Activate]
+        [HtmlAttributeNotBound]
+        public ViewContext ViewContext { get; set; }
+
+        [Activate]
+        [HtmlAttributeNotBound]
+        public IMemoryCache Cache { get; set; }        
+
+         /// <inheritdoc />
+        public override void Process(TagHelperContext context, TagHelperOutput output)
+        {
+            if (Src != null)
```

Comment:
With https://github.com/aspnet/Mvc/pull/2516/files#r30006170 you wont need to check for null here because it'll always be provided. Also wont need to restore the src attribute on line 71.